storage/queue.go

```diff
@@ -111,6 +112,96 @@ type PeekMessageResponse struct {
 	MessageText    string `xml:"MessageText"`
 }
 
+// QueueMetadata represents user defined metadata and queue properties on
+// a specific queue.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179384.aspx
+type QueueMetadata struct {
+	ApproximateMessageCount int
+	userDefinedMetadata     map[string]string
```

Comment:
for now I kept it as-is (the way to golang canonical works) and documented it. lmk if its fine or you prefer lowercasing